src/rgw/rgw_admin.cc

```diff
@@ -2575,6 +2575,13 @@ int check_reshard_bucket_params(rgw::sal::RGWRadosStore *store,
     return ret;
   }
 
+  if (bucket_info.reshard_status != cls_rgw_reshard_status::NOT_RESHARDING) {
```

Comment:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc: In function 'int check_reshard_bucket_params(rgw::sal::RGWRadosStore*, const string&, const string&, const string&, bool, int, int, rgw_bucket&, RGWBucketInfo&, std::map<std::__cxx11::basic_string<char>, ceph::buffer::v14_2_0::list>&)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc:2578:61: error: 'NOT_RESHARDING' is not a member of 'cls_rgw_reshard_status'
   if (bucket_info.reshard_status != cls_rgw_reshard_status::NOT_RESHARDING) {
                                                             ^~~~~~~~~~~~~~
src/rgw/CMakeFiles/radosgw-admin.dir/build.make:62: recipe for target 'src/rgw/CMakeFiles/radosgw-admin.dir/rgw_admin.cc.o' failed
```

can we `#include "cls/rgw/cls_rgw_types.h"` in this source file as well?

Reply:
I did this in an other PR -- https://github.com/ceph/ceph/pull/30610, and it is now in master.